inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/tasks/SelectionTask.java

```diff
@@ -177,23 +184,56 @@ public void run()
                                 user.getUsername(), recommenderName, score,
                                 threshold);
                     }
-
-                    appEventPublisher.publishEvent(new RecommenderEvaluationResultEvent(this,
-                            recommender, user.getUsername(), result,
-                            System.currentTimeMillis() - start, activated));
+                    publishEvalEvent(user, recommenders.size(), recommenderCount, recommender,
+                            start, result, activated);
+                    
                 }
                 catch (Throwable e) {
-                    log.error("[{}][{}]: Failed", user.getUsername(), recommenderName, e);
+                    log.error("[{}][{}]: Failed", userName, recommenderName, e);
+                    appEventPublisher
```

Comment:
Is it correct to send a `TaskState.DONE` here? The loop still continues, even if an individual recommender fails.